app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/EnrolmentAlreadyExistsController.scala

```diff
@@ -35,16 +35,14 @@ class EnrolmentAlreadyExistsController @Inject() (
   mcc: MessagesControllerComponents
 ) extends FrontendController(mcc) with I18nSupport {
 
-  // Note: permitted for user with service enrolment
   def enrolmentAlreadyExists(service: Service): Action[AnyContent] =
-    authAction.ggAuthorisedUserWithServiceAction {
+    authAction.ggAuthorisedUserAction {
```

Comment:
Why is this swapped over?

Reply:
Good question - `ggAuthorisedUserWithServiceAction` is implemented in such way that it removes all enrolments  which causes enrolment check to fail and in turn redirecting again back to `enrolmentAlreadyExists` page.

 So basically if this is left with `ggAuthorisedUserWithServiceAction ` - when user gets redirected from Start page to `enrolmentAlreadyExists` it would end up in infinite loop redirecting to the same `enrolmentAlreadyExists` page.

By checking commit and ticket history is not obvious why it is implemented like this. Before the change all was working  because the check for Assistant type was done first ant that resulted redirecting to a different page.